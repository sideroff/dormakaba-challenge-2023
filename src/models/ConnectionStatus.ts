export const ConnectionStatus = {
  Online: 'online',
  Offline: 'offline',
} as const;

export type ConnectionStatus =
  (typeof ConnectionStatus)[keyof typeof ConnectionStatus];
