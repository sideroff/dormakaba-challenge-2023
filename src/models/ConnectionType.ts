export const ConnectionType = {
  Wired: 'wired',
  Wireless: 'wireless',
} as const;

export type ConnectionType =
  (typeof ConnectionType)[keyof typeof ConnectionType];
