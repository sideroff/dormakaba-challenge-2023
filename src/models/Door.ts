import { ConnectionType } from './ConnectionType';
import { ConnectionStatus } from './ConnectionStatus';

export interface Door {
  id: string;
  name: string;
  buildingName: string;
  connectionType: ConnectionType;
  connectionStatus: ConnectionStatus;
  lastConnectionStatusUpdate: string;
}
