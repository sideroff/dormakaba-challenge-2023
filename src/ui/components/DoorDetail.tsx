import Typography from '@mui/material/Typography';
import { Door } from '@/models/Door';
import { DetailPageContainer } from '@/ui/layout/DetailPageContainer';
import { DetailPageItem } from '@/ui/layout/DetailPageItem';
import ConnectionStatusLabel from './ConnectionStatusLabel';
import { getLocaleString } from '@/lib/dateTime';

interface DoorDetailProps {
  door: Door;
}

export function DoorDetail({ door }: DoorDetailProps) {
  return (
    <DetailPageContainer>
      <DetailPageItem label="ID">
        <Typography>{door.id}</Typography>
      </DetailPageItem>
      <DetailPageItem label="Building">
        <Typography>{door.buildingName}</Typography>
      </DetailPageItem>
      <DetailPageItem label="Apartment">
        <Typography>{door.apartmentName}</Typography>
      </DetailPageItem>
      <DetailPageItem label="Connection type">
        <Typography>{door.connectionType}</Typography>
      </DetailPageItem>
      <DetailPageItem label="Connection status">
        <ConnectionStatusLabel connectionStatus={door.connectionStatus} />
      </DetailPageItem>
      <DetailPageItem label="Last connection status update">
        <Typography>
          {getLocaleString(door.lastConnectionStatusUpdate)}
        </Typography>
      </DetailPageItem>
    </DetailPageContainer>
  );
}
