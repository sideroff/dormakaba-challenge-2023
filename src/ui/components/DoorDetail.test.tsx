import { render } from '@testing-library/react';
import { Door } from '@/models/Door';
import { DoorDetail } from './DoorDetail';

const door: Door = {
  id: '63f637c9f3c48a124616044b',
  name: 'Building Main Entrance',
  buildingName: 'Bahnhofstrasse 10A',
  connectionType: 'wired',
  connectionStatus: 'offline',
};

describe('DoorDetail', () => {
  it('should render correctly', () => {
    const { container } = render(<DoorDetail door={door} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
