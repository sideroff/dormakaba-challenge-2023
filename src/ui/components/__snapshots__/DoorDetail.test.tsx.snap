// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoorDetail should render correctly 1`] = `
<div
  class="MuiStack-root css-pwj874-MuiStack-root"
>
  <div
    class="MuiBox-root css-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      <strong>
        ID
      </strong>
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      63f637c9f3c48a124616044b
    </p>
  </div>
  <div
    class="MuiBox-root css-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      <strong>
        Building
      </strong>
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Bahnhofstrasse 10A
    </p>
  </div>
  <div
    class="MuiBox-root css-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      <strong>
        Apartment
      </strong>
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    />
  </div>
  <div
    class="MuiBox-root css-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      <strong>
        Connection type
      </strong>
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      wired
    </p>
  </div>
  <div
    class="MuiBox-root css-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      <strong>
        Connection status
      </strong>
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1oykq5t-MuiTypography-root"
    >
      offline
    </p>
  </div>
  <div
    class="MuiBox-root css-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      <strong>
        Last connection status update
      </strong>
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    />
  </div>
</div>
`;
